### PR TITLE
Remove numpy #1

### PR DESCRIPTION
**Pull request by imaolo**
- [x] Remove numpy from the dtypes (it is just a string now)
- [x] Support Tensor creation from lists and nested lists without bumpy
- [ ] Add some clean method to get the Tensor out as a native Python object

Using ctypes for the canonical conversions, but there is no float16 ctype so some manual conversion will need to happen eventually.

to_mv/frombuffer has replaced np.array

 #2649

should probably consider using struct.pack()

### CODE REVIEW

File: tinygrad/helpers.py [hunk 1]
```diff
@@ -143,25 +163,29 @@ def is_float(x: DType) -> bool: return x in (dtypes.float16, dtypes.float32, dty
...
+  int8: Final[DType] = DType(1, 1, "char", "int8", ctypes.c_byte)
+  int16: Final[DType] = DType(3, 2, "short", "int16", ctypes.c_short)
+  int32: Final[DType] = DType(5, 4, "int", "int32", ctypes.c_int)
   int = int32
-  int64: Final[DType] = DType(7, 8, "long", np.int64)
-  uint8: Final[DType] = DType(2, 1, "unsigned char", np.uint8)
-  uint16: Final[DType] = DType(4, 2, "unsigned short", np.uint16)
-  uint32: Final[DType] = DType(6, 4, "unsigned int", np.uint32)
-  uint64: Final[DType] = DType(8, 8, "unsigned long", np.uint64)
+  int64: Final[DType] = DType(7, 8, "long", "int64", ctypes.c_long)
```
Comment (geohot) [inline tinygrad/helpers.py:184]: Why are these strings here?

**Comment by geohot** [comment]
This is still too complex for a single PR. If the title of the PR is "remove numpy #1" and not a clear single "feature", it won't be merged

**Comment by imaolo** [comment]
> This is still too complex for a single PR. If the title of the PR is "remove numpy #1" and not a clear single "feature", it won't be merged

We need the strings so we can support numpy(). They act the same when used in np.array(dtype=) but are not np.dtypes().

Will work on removing them.

File: tinygrad/helpers.py (hunk 1 above)
Comment (imaolo) [reply tinygrad/helpers.py:184]: We need the strings so we can support numpy(). They act the same when used in np.array(dtype=) but are not np.dtypes().

Will work on removing them.

**Pull request closed without merging**